### ISSUE TITLE
Unsupported attribute axis of type Softmax

### ISSUE THREAD

**Issue by mozart20** [Unity-Technologies/barracuda-release#48]
I have a very simple CNN model trained in Keras, converted to onnx. I have verified that the Keras to onnx conversion worked fine and the outputs are the same. But when I load it into Unity, I get all sorts of cryptic warnings like "Unsupported attribute axis, node dense of type Softmax" or "Unexpected error while parsing layer... of type Transpose." And the output values are totally different.

I thought Barracuda supported Softmax? Why am I getting these warnings, and why is the inference incorrect?

These are the layers in my model.

network = models.Sequential()
network.add(layers.Conv2D(32, (3, 3), activation='relu', input_shape=(1, image_height, image_width)))
network.add(layers.MaxPooling2D((2, 2)))
network.add(layers.Conv2D(64, (3, 3), activation='relu'))
network.add(layers.MaxPooling2D((2, 2)))
network.add(layers.Conv2D(64, (3, 3), activation='relu'))
network.add(layers.MaxPooling2D((2, 2)))
network.add(layers.Conv2D(64, (3, 3), activation='relu'))
network.add(layers.MaxPooling2D((2, 2)))
network.add(layers.Conv2D(64, (3, 3), activation='relu'))
network.add(layers.MaxPooling2D((2, 2)))
network.add(layers.Flatten())
network.add(layers.Dense(516, kernel_regularizer=regularizers.l2(0.001), activation='relu'))
network.add(layers.Dense(num_classes, activation='softmax'))

Thanks!

**Comment by mantasp** [comment]
Transpose layers are currently not fully supported in Barracuda, we are working on it. As temporary workaround you might try loading Keras model in TF2.x and then freezing it and saving as .pb, which later can be converted to .onnx via tf2onnx.
Here is .h5 -> .pb sample conversion code that worked for me:
```
import logging
import tensorflow as tf
from tensorflow.compat.v1 import graph_util
from tensorflow.python.keras import backend as K
from tensorflow import keras
# necessary !!!
tf.compat.v1.disable_eager_execution()
K.set_learning_phase(0)
h5_path = 'mnist-cnn-mini.h5'
model = keras.models.load_model(h5_path)
model.summary()
# save pb
with K.get_session() as sess:
    output_names = [out.op.name for out in model.outputs]
    input_graph_def = sess.graph.as_graph_def()
    for node in input_graph_def.node:
        node.device = ""
    graph = graph_util.remove_training_nodes(input_graph_def)
    graph_frozen = graph_util.convert_variables_to_constants(sess, graph, output_names)
    tf.io.write_graph(graph_frozen, '.', 'mnist-cnn-mini.pb', as_text=False)
logging.info("save pb successfully！")
```

**Comment by JackWei0310** [comment]
Yes. It works now. Thanks for helping!
